### PR TITLE
(2533) Create Level B activities from bulk upload CSV

### PR DESCRIPTION
**Pull request by yndajas**
## Changes in this PR

This adds an update action to the Level B activity uploads controller, which handles uploads in the same way as the existing activity uploads controller

### Full run through/failure

https://user-images.githubusercontent.com/40244233/190205088-6e50da31-bdb1-43aa-be9d-be4865355734.mov

### Success screen

image

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

File: spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb [hunk 1]
```diff
@@ -44,4 +42,155 @@
...
+    expect(activity_to_update.reload.title).to eq("New Title")
+
+    within "//tbody/tr[1]" do
+      expect(page).to have_xpath("td[2]", text: "New Title")
+    end
+  end
+
+  scenario "attempting to change the partner organisation identifier of an existing activity" do
+    activity_to_update = create(:project_activity, :gcrf_funded, organisation: organisation)
+    # create(:report, :active, fund: activity_to_update.associated_fund, organisation: organisation)
```
Comment (CristinaRO) [inline spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb:151]: Commented line - is it to be removed?

File: spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb [hunk 2]
```diff
@@ -44,4 +42,155 @@
...
+      expect(page).to have_xpath("td[1]", text: "Free Standing Technical Cooperation")
+      expect(page).to have_xpath("td[2]", text: "3")
+      expect(page).to have_xpath("td[3]", text: "")
+      expect(page).to have_xpath("td[4]", text: t("activerecord.errors.models.activity.attributes.fstc_applies.inclusion"))
+    end
+  end
+
+  scenario "updating an existing activity" do
+    activity_to_update = create(:fund_activity, :newton)
+    # create(:report, :active, fund: activity_to_update.associated_fund, organisation: organisation)
```
Comment (CristinaRO) [inline spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb:118]: Commented line.

File: spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb [hunk 3]
```diff
@@ -44,4 +42,155 @@
...
+  scenario "uploading an empty file" do
+    upload_csv(Activities::ImportFromCsv.column_headings.join(", "))
+
+    expect(page).to have_text(t("action.activity.upload.file_missing_or_invalid"))
+  end
+
+  scenario "uploading a valid set of activities" do
+    old_count = Activity.count
+
+    # When I upload a valid Activity CSV with comments
```
Comment (CristinaRO) [inline spec/features/staff/beis_users_can_upload_level_b_activities_spec.rb:61]: Did we decide to remove comments from this version?

**Review by CristinaRO — APPROVED** [review]
Approved with a few comments/requests.

**Review by Gweaton — APPROVED** [review]
I've done some manual testing and can confirm that:
- BEIS users can't upload level C activities using this form
- Level C activities can be added against bulk-created level B activities via the Level C and D bulk upload
- The bug we saw yesterday was caused by something we forgot to rename in the renaming branch, and is unrelated to this change.

**Pull request merged**